infrastructure/crypto/src/keys.rs

```diff
@@ -29,6 +30,7 @@ pub trait SecretKeyFactory {
 /// ```
 pub trait SecretKey {
     fn key_length() -> usize;
+    fn to_scalar(&self) -> Scalar;
```

Comment:
Leaky abstraction? What if we want to change to say, secp256k1?

Reply:
Of course, I should just have used the already provided `impl From<RistrettoSecretKey> for Scalar` implementation for this functionality. Fixed.